rune/libenclave/internal/runtime/pal/skeleton/sgxsign.c

```diff
@@ -153,19 +155,34 @@ struct mrecreate {
 /* *INDENT-OFF* */
 static bool mrenclave_ecreate(EVP_MD_CTX *ctx, uint64_t blob_size,
 			      uint32_t miscselect, uint64_t xfrm,
-			      uint32_t *ssa_frame_size)
+			      uint32_t *ssa_frame_size, uint64_t memory_size)
 /* *INDENT-ON* */
 {
 	struct mrecreate mrecreate;
 	uint64_t encl_size;
 
-	for (encl_size = PAGE_SIZE; encl_size < blob_size;)
-		encl_size <<= 1;
-
 	memset(&mrecreate, 0, sizeof(mrecreate));
 	mrecreate.tag = MRECREATE;
 	mrecreate.ssaframesize = sgx_calc_ssaframesize(miscselect, xfrm);
 	*ssa_frame_size = mrecreate.ssaframesize;
+
+	if (set_memory_size) {
```

Comment:
Redundant.

If -m option is specified, the argument will be a valid value, and thus memory_size itself can indicate whether it is specified, without the assist of set_memory_size.